### ISSUE TITLE
Error when compiling a file using glob with perl 5.20

### ISSUE THREAD

**Issue by atoomic** [rurban/perl-compiler#223]
Env:
perl 5.20.2
B::C using bc522 with HEAD @734e776 ( cannot use 23016e1 with 5.20 )

This is a regression as the same test is not generating any warnings when using perl 5.14.4
and it's passing

````
> cat test.pl
#!./perl -w

my @boom = glob('*');
print "ok - no warnings\n"; # we should also check that no warnings are displayed
> perl test.pl
ok - no warnings
> perlcc -r test.pl
````

different errors when using a cm520 branch
````
Constant subroutine GLOB_ABEND redefined at ./test line 4294967295.
Constant subroutine GLOB_ALPHASORT redefined at ./test line 4294967295.
Constant subroutine GLOB_ALTDIRFUNC redefined at ./test line 4294967295.
Constant subroutine GLOB_BRACE redefined at ./test line 4294967295.
Constant subroutine GLOB_ERR redefined at ./test line 4294967295.
Constant subroutine GLOB_LIMIT redefined at ./test line 4294967295.
Constant subroutine GLOB_MARK redefined at ./test line 4294967295.
Constant subroutine GLOB_NOCASE redefined at ./test line 4294967295.
Constant subroutine GLOB_NOCHECK redefined at ./test line 4294967295.
Constant subroutine GLOB_NOMAGIC redefined at ./test line 4294967295.
Constant subroutine GLOB_NOSORT redefined at ./test line 4294967295.
Constant subroutine GLOB_NOSPACE redefined at ./test line 4294967295.
Constant subroutine GLOB_QUOTE redefined at ./test line 4294967295.
Constant subroutine GLOB_TILDE redefined at ./test line 4294967295.
Constant subroutine GLOB_CSH redefined at ./test line 4294967295.
````

**Comment by rurban** [comment]
I cannot repro that at all. Still relevant?

**Comment by atoomic** [comment]
this is still an issue with HEAD@da5b30c3b and using perl 5.22

**Comment by atoomic** [comment]
@rurban easy way to reproduce the problem
````
> perl -e 'glob "wtf.*"; print "ok\n"'
ok
> perlcc -r -e 'glob "wtf.*"; print "ok\n"'
Segmentation fault
````

**Comment by toddr** [comment]
Confirmed reproducible

**Comment by atoomic** [comment]
another one
````
> perl -e '<*.*> and print qq{ok\n}'
ok
> perlcc -r -e '<*.*> and print qq{ok\n}'
Segmentation fault
````
